### PR TITLE
Adds additional capabilities to the minimal JS wrapper

### PR DESCRIPTION
**Pull request by greglandrum**
what's in this PR:

- `get_substruct_match()` now returns matching atoms and bonds
- `get_substruct_matches()`
- a set of functions that should (I hope) allow the JS code to be used by ketcher as an alternative to a server.

### CODE REVIEW

File: Code/MinimalLib/minilib.cpp [hunk 1]
```diff
@@ -72,19 +73,32 @@ ROMol *qmol_from_input(const std::string &input) {
...
     highlight_atoms->reserve(atomIds->size());
     for (auto ai : *atomIds) {
       highlight_atoms->push_back(ai);
     }
   }
-  MolDraw2DUtils::prepareAndDrawMolecule(drawer, m, "", highlight_atoms);
+
+  std::vector<int> *highlight_bonds = nullptr;
+  if (bondIds && bondIds->size()) {
+    highlight_bonds = new std::vector<int>;
```
Comment (bp-kelley) [inline Code/MinimalLib/minilib.cpp]: Looks like a good place to get into the habit of using a unique_ptr.

File: Code/MinimalLib/minilib.cpp [hunk 2]
```diff
@@ -113,7 +127,16 @@ std::string JSMol::get_svg_with_highlights(const std::string &details) const {
     if (!molval.IsInt()) return ("Atom IDs should be integers");
     atomIds.push_back(static_cast<unsigned int>(molval.GetInt()));
   }
-  return svg_(*d_mol, &atomIds);
+  std::vector<unsigned int> bondIds;
+  if (doc.HasMember("bonds") && !doc["bonds"].IsArray()) {
+    return "JSON doesn't contain 'bonds' field, but it is not an array";
```
Comment (bp-kelley) [inline Code/MinimalLib/minilib.cpp]: Why not throw an exception?  This looks like the user has to decide whether it is a valid SVG or not.

File: Code/MinimalLib/minilib.cpp [hunk 3]
```diff
@@ -179,6 +248,96 @@ std::string JSMol::get_morgan_fp(unsigned int radius,
...
+
+  bool cleanIt = true;
+  bool force = true;
+  MolOps::assignStereochemistry(*d_mol, cleanIt, force);
+
+  rj::Value rjAtoms(rj::kArrayType);
+  for (const auto atom : d_mol->atoms()) {
+    std::string cip;
+    if (atom->getPropIfPresent(common_properties::_CIPCode, cip)) {
+      cip = "(" + cip + ")";
```
Comment (bp-kelley) [inline Code/MinimalLib/minilib.cpp]: Out of curiosity, why are these surrounded by parentheses?

File: Code/MinimalLib/minilib.cpp (hunk 2 above)
Comment (bp-kelley) [reply Code/MinimalLib/minilib.cpp]: Right, emscripten doesn't really support them.

Comment (greglandrum) [reply Code/MinimalLib/minilib.cpp]: yep. And, realistically this shouldn't happen anyway

File: Code/MinimalLib/minilib.cpp (hunk 3 above)
Comment (greglandrum) [reply Code/MinimalLib/minilib.cpp]: That's what the current ketcher services do, apparently

**Pull request merged**